.github/workflows/Tests.yaml

```diff
@@ -433,3 +433,87 @@ jobs:
           # We get occasional random timeouts, repeat tests to see if
           # it is a random timeout or systematic
           ctest -j2 --output-on-failure -L inputfiles --repeat after-timeout:3
+
+  arch_datastructures_tests:
+    name: Archs
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        compiler:
+          - "gcc-7"
+          - "gcc-9"
+          - "clang-6.0"
+          - "clang-10"
```

Comment:
Do you think this is the smallest possible number of CI builds, or could we reduce the number of compiler configurations further to speed up CI? Related to this comment https://github.com/sxs-collaboration/spectre/pull/2574/#discussion_r515477632:
> Do we have specific cluster/compiler/arch combinations in production that would be important to test, e.g. the case that prompted the revert?

Reply:
The answer here is that it is not terribly clear. The single case we have encountered is in the most modern compiler, with one of the latest architectures, so that tends to indicate that 'new is buggy', but I could also see an argument that the combination of oldest compiler with newest vectorization flags could be concerning. So, I've picked the oldest and newest supported compilers for each of clang and gcc, but the pick is pretty arbitrary. It at least certainly covers the single case that failed previously.